File: BOJ/FastCampas_Algorithm_lec/BOJ_2798/BOJ_2798_2.cpp
#include<iostream>
using namespace std;
int max(int a, int b) {
	return a > b ? a : b;
}

int main() {
	int N, M, res = 0, ans = 0;
	int card[101];
	cin >> N >> M;
	for (int i = 0; i < N; i++) {
		cin >> card[i];
	}

	for (int i = 0; i < N - 2; i++) {
		for (int j = i + 1; j < N - 1; j++) {
			for (int k = j + 1; k < N; k++) {
				res = card[i] + card[j] + card[k];
				if (res <= M) {
					ans = max(res, ans);
				}
			}
		}
	}
	cout << ans << "\n";
	return 0;
}

File: BOJ/FastCampas_Algorithm_lec/BOJ_2798/BOJ_2798_2.py
n, m = list(map(int, input().split(' ')))
data = list(map(int, input().split(' ')))

res = 0
length = len(data)

for i in range(0, length):
    for j in range(i + 1, length):
        for k in range(j + 1, length):
            sumvalue = data[i] + data[j] + data[k]
            if sumvalue <= m:
                res = max(res, sumvalue)
print(res)
